glib/Gir.toml

```diff
@@ -51,17 +51,38 @@ manual = [
 name = "GLib.*"
 status = "generate"
     [[object.function]]
-    name = "clear_error"
-    #unusable
-    ignore = true
+    pattern = "(assert_warning|assertion_message|assertion_message_cmpnum|assertion_message_cmpstr|warn_message|return_if_fail_warning)"
+    ignore = true # Not useful assertions functions
     [[object.function]]
-    name = "propagate_error"
-    #empty first parameter
-    ignore = true
+    pattern = "(set_print_handler|set_printerr_handler|vasprintf|vfprintf|vprintf|vsnprintf|vsprintf|stpcpy|sprintf|snprintf|printf|printf_string_upper_bound|printerr|prefix_error|print|fprintf)"
+    ignore = true # Not useful string functions
     [[object.function]]
-    name = "set_error_literal"
-    #empty first parameter
-    ignore = true
+    pattern = "(malloc|malloc0|malloc0_n|malloc_n|memdup|memdup2|realloc|realloc_n|try_malloc||try_malloc0|try_malloc0_n|try_malloc_n|try_realloc|try_realloc_n|free)"
+    ignore = true # Not useful allocations functions
+    [[object.function]]
+    pattern = "(slice_get_config|slice_get_config_state|slice_set_config|slice_alloc|slice_alloc0|slice_copy|slice_free1|slice_free_chain_with_offset)"
+    ignore = true # Not useful slices functions
+    [[object.function]]
+    pattern = "(path_is_absolute|build_filename|build_filename_valist|build_filenamev|build_path|build_pathv|canonicalize_filename|chdir)"
+    ignore = true # Not useful paths functions, use Path/PathBuf
+    [[object.function]]
+    pattern = "(int64_equal|int64_hash|int_equal|int_hash|double_equal|double_hash|direct_equal|direct_hash)"
+    ignore = true # Not useful equal/hash functions, only useful for GHashTable
+    [[object.function]]
+    pattern = "(clear_handle_id|clear_list|clear_pointer|clear_slist|nullify_pointer)"
+    ignore = true # Not useful freeing functions
+    [[object.function]]
+    pattern = "(datalist_.*|dataset_.*)"
+    ignore = true # Not useful GData related functions
+    [[object.function]]
+    pattern = "(propagate_prefixed_error|set_error|file_error_from_errno|clear_error|propagate_error|set_error_literal|set_error)"
+    ignore = true # Not useful error handling functions
+    [[object.function]]
+    pattern = "(idle_remove_by_data|qsort_with_data)"
+    ignore = true # Unsafe functions
+    [[object.function]]
+    pattern = "(iconv|convert_with_iconv)"
+    ignore = true # not usable
     [[object.function]]
     pattern = "atomic_.+"
     ignore = true
```

Comment:
Other functions
 - `g_access` (that's in `std` already)
 - `g_base64_decode` (this should really be manual and return a `glib::Slice<u8>` that's just silly otherwise)
 - `g_base64_.*_(step|inplace|close)` (these need manual bindings if we want such a thing)
 - `g_bit_*` (`std`)
 - `g_compute_.*_for_string` (should be manually implemented because this one just silly)
 - `g_file_open_tmp` (the returned `fd` should be used with the corresponding type from `std`)
 - `g_file_read_link` (`std`)
 - `g_file_test` (`std`)
 - `g_get_current_time` (deprecated and there's a better function)
 - `g_io_.*` (needs manual bindings but it's also awful API)
 - `g_log_.*` (needs manual bindings if we want those to be useful)
 - `g_markup_.*` (these are together with `GMarkupParser` which we don't generate)
 - `g_parse_debug_string` (needs manual bindings but also rather useless)
 - `g_pattern_.*` (pattern matching functions, we have a better API for that in `std`)
 - `g_pointer_.*` (raw pointers...)
 - `g_poll` (manual bindings but there are crates doing this so...)
 - `g_rmdir` (`std`)

While we're at it, also please generate the `GUnicodeScript` enum then we get a few more useful functions here.

Reply:
- [x] `g_access` (that's in `std` already)
 - `g_base64_decode` (this should really be manual and return a `glib::Slice<u8>` that's just silly otherwise)
 - `g_base64_.*_(step|inplace|close)` (these need manual bindings if we want such a thing)

Regarding g_base64, those are rather useful as is, they involve unnecessary copies but they are not fundamentally broken. So I guess let's keep them until someone is interested in improving them in the future?

 - [x] `g_bit_*` (`std`)
 - `g_compute_.*_for_string` (should be manually implemented because this one just silly)

What's not great on the current implementation as is? 

 - [x] `g_file_open_tmp` (the returned `fd` should be used with the corresponding type from `std`)
 
 I've manually implemented it for unix only, no idea how that would look like for windows.
 
 - [x] `g_file_read_link` (`std`)
 - [x] `g_file_test` (`std`)
 - [x] `g_get_current_time` (deprecated and there's a better function)
 - [x] `g_io_.*` (needs manual bindings but it's also awful API)
 - `g_log_.*` (needs manual bindings if we want those to be useful)
 
 As there's a possibility to make a nicer wrapper around them with manual bindings, let's leave those as is so they are still generated as "ignored" on the code?
 
 - [x] `g_markup_.*` (these are together with `GMarkupParser` which we don't generate)
 - `g_parse_debug_string` (needs manual bindings but also rather useless)

 Same as others that require manual bindings

 - [x] `g_pattern_.*` (pattern matching functions, we have a better API for that in `std`)
 - [x] `g_pointer_.*` (raw pointers...)
 
 - `g_poll` (manual bindings but there are crates doing this so...)
 
 Would leave this as is, in case someone wants to fix it someday

 - [x] `g_rmdir` (`std`)

Reply:
> What's not great on the current implementation as is?

`g_compute_.*_for_string` creates a temporary copy of the string in the bindings, while `compute_for_data(a_string.as_bytes())` wouldn't. Having the string function there is misleading because it looks equally cheap but definitely isn't.

> I've manually implemented it for unix only, no idea how that would look like for windows.

I think `std` has no fd API on Windows, you'd have to go through `HANDLE` somehow.

> As there's a possibility to make a nicer wrapper around them with manual bindings, let's leave those as is so they are still generated as "ignored" on the code?

You're generating some of the logging functions but they're useless without the remainder

> Same as others that require manual bindings

Well, `g_parse_debug_string` is also useless :)

Reply:
`GFileTest` enum can be ignored now

Reply:
@bilelmoussaoui What's the status here?

Reply:
Will get back to it in a bit, finishing another "cleanup" merge request for gtk-rs-core first :)

Reply:
> GFileTest enum can be ignored now

Done

> Well, g_parse_debug_string is also useless :)

Dropped

> I think std has no fd API on Windows, you'd have to go through HANDLE somehow.

As I've no idea how anything windows related works, I will leave that to someone that needs it.

> You're generating some of the logging functions but they're useless without the remainder

Done